Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -799,6 +808,22 @@ private extension PurchasesOrchestrator {
             #endif
         }
     }
+
+    private func initiationSource(
+        for productIdentifier: String,
+        restored: Bool
+    ) -> ProductRequestData.InitiationSource {
+        // Having a purchase completed callback implies that the transation comes from an explicit call
+        // to `purchase()` instead of a StoreKit transaction notification.
```

Comment:
I think this works.

I got quite confused at first, since that restore isn't coming from us, and we do offer the ability to attach a completion block to restores when they get initiated through our SDK. 

But restores that _are_ coming from us won't actually go through this flow in the first place since they don't issue transactions in the queue. Instead, for our restores we just refresh the receipt and send it. 

Maybe we can document that behavior more in detail?